logging/error.go

```diff
@@ -38,13 +39,32 @@ func FailValidation(root, details error) ValidationError {
 	return v
 }
 
+func FailNamespacedValidation(root, err error, namespace string) error {
```

Comment:
I didn't port the `Namespace` as it's seemingly not necessary - the frontend can manage just fine without specifying it in our errors. It can be removed from this library so that API of this package is kept simple.